test/screenshot/lib/controller.js

```diff
@@ -252,17 +252,109 @@ class Controller {
       .then(
         (image) => {
           return new Promise((resolve, reject) => {
+            const {rows, cols} = this.getCropMatches_(image);
+            const cropRect = this.getCropRect_({rows, cols});
+            const {x, y, w, h} = cropRect;
+
             image
-              .autocrop()
+              .crop(x, y, w, h)
               .getBuffer(jimp.MIME_PNG, (err, buffer) => {
                 return err ? reject(err) : resolve(buffer);
-              });
+              })
+            ;
           });
         },
         (err) => Promise.reject(err)
       );
   }
 
+  getCropMatches_(image) {
+    const trimColors = [0x333333FF];
+    const rows = [];
+    const cols = [];
+
+    image.scan(0, 0, image.bitmap.width, image.bitmap.height, (x, y) => {
+      if (!rows[y]) {
+        rows[y] = [];
+      }
+      if (!cols[x]) {
+        cols[x] = [];
+      }
+      const isMatch = trimColors.includes(image.getPixelColor(x, y));
+      rows[y][x] = isMatch;
+      cols[x][y] = isMatch;
+    });
+
+    return {rows, cols};
+  }
+
+  getCropRect_({rows, cols}) {
+    const HIGH_MATCH_PERCENTAGE = 0.95;
+    const LOW_MATCH_PERCENTAGE = 0.67;
+
+    const amounts = {
+      top: 0,
+      left: 0,
+      right: 0,
+      bottom: 0,
+    };
+
+    for (const row of rows) {
+      if (matchPercentage(row) > HIGH_MATCH_PERCENTAGE) {
+        amounts.top++;
+      } else {
+        break;
+      }
+    }
+
+    for (const row of rows.concat().reverse()) {
```

Comment:
Are you using `.concat()` as a cheap array cloning method since `.reverse()` modifies the array in-place?

Reply:
Yes, exactly! Should I add a comment explaining that?

Reply:
Yeah, a one-liner is fine.